gnocchi/storage/redis.py

```diff
@@ -114,16 +128,18 @@ def _delete_metric(self, metric):
     def _get_measures(self, metric, keys, aggregation, version=3):
         if not keys:
             return []
-        redis_key = self._metric_key(metric)
         fields = [
             self._aggregated_field_for_split(aggregation, key, version)
             for key in keys
         ]
-        results = self._client.hmget(redis_key, fields)
-        for key, data in six.moves.zip(keys, results):
-            if data is None:
-                if not self._client.exists(redis_key):
-                    raise storage.MetricDoesNotExist(metric)
-                raise storage.AggregationDoesNotExist(
-                    metric, aggregation, key.sampling)
-        return results
+        code, result = self._scripts['get_measures'](
+            keys=[self._metric_key(metric)],
+            args=fields,
+        )
+        if code == -1:
+            sampling = utils.to_timespan(result.split(self.FIELD_SEP_B)[2])
+            raise storage.AggregationDoesNotExist(
+                metric, aggregation, sampling)
+        if code == -2:
+            raise storage.MetricDoesNotExist(metric)
```

Comment:
i don't think we have tests for this as i deleted the following and it still passes.

```
+        if code == -1:
+            sampling = utils.to_timespan(result.split(self.FIELD_SEP_B)[2])
+            raise storage.AggregationDoesNotExist(
+                metric, aggregation, sampling)
+        if code == -2:
+            raise storage.MetricDoesNotExist(metric)
```

tbh, i'm not sure it's possible to get this far as we do validate the aggregation beforehand.

Reply:
I just kept the same structure, so IIUC, this is not a new problem (if it is one at all)?

Reply:
well it's a problem in that the errors raise are never validated so in theory, you could've broken it :P